Add support for acceleration sensor in Samsung Smartthings Multipurpo…

robstonham-personal opened the pull request:
…se sensor (2016 model)

Add the attribute that indicates the device has been moved.

I'm not sure sure how to add this purely in Zigbee-herdsman-converters, so needed to add the following to the Zigbee-herdsman package as well.


file: src/zcl/definition/manufacturerCode.ts:
export default {
    Centralite: 0x104E,
    Philips: 0x100B,
    Sinope: 0x119C,
    Stelpro: 0x1185,
    Ubisys: 0x10f2,
    LegrandNetatmo: 0x1021,
    SmartThings: 0x110a,
};

and
file: src/zcl/definition/cluster.ts:
    manuSpecificSamsungAccelerometer: {
        ID: 0xFC02,
        manufacturerCode: ManufacturerCode.SmartThings,
        attributes: {
             motion_threshold_multiplier: {ID: 0, type: DataType.uint8},
             motion_threshold: {ID: 2, type: DataType.uint16},
             acceleration: {ID: 16, type: DataType.bitmap8},
             x_axis: {ID:18, type: DataType.int16},
             y_axis: {ID:19, type: DataType.int16},
             z_axis: {ID:20, type: DataType.int16},
        },
        commands: {},
        commandsResponse: {},
    },

Koenkk commented on converters/fromZigbee.js:
```diff
@@ -5050,6 +5050,15 @@ const converters = {
+        cluster: 'manuSpecificSamsungAccelerometer',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            return {acceleration: msg.data['acceleration'] === 1 ? true : false};
```
Is this a state or an event? To me it looks like an event if that is indeed the case you can change this to `return {action: 'acceleration'};`

robstonham-personal replied:
To me it's a state. The device sends a message when I pickup the device, and then sends another message after it stops moving for a few seconds.

Koenkk replied:
Clear! In that case `accelerating` or `moving` would be a better name

robstonham-personal replied:
Sorry for the delay, been away.
I've updated to use "moving" as suggested

dejanzelic commented:
+1 for support of this! I have a 2016 F-MLT-US-2 if we need to test anything

Koenkk commented on converters/fromZigbee.js:
```diff
@@ -5050,6 +5050,15 @@ const converters = {
+        cluster: 'manuSpecificSamsungAccelerometer',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            return {acceleration: msg.data['moving'] === 1 ? true : false};
```
I guess you meant to change this to?

```js
return {moving: msg.data['acceleration'] === 1 ? true : false};
```

robstonham-personal replied:
yep, so sorry, fixed it as suggested

Koenkk commented:
Thanks!

dejanzelic commented:
@robstonham-personal were you able to confirm this as working? I have 2 Samsung Smartthings v4 multisensors that never toggle the "moving" attribute. The moving attribute is recongnised on the devices (and shows in mqtt and home assistant) but moving is never triggered to be true (no matter how much I move the sensor). I know that these sensors have the accelerometer because they used to work with ZHA.

robstonham-personal commented:
Yep I had it working last time I looked at it.  After I updated though I
found that it stopped working for me too.
My logs showed this though:
Zigbee2MQTT:debug 2020-08-23 23:04:12: No converter available for
'F-MLT-US-2' with cluster 'manuSpecificSamsungAccelerometer' and type
'attributeReport' and data '{"acceleration":1}'

I just added "fz.smartthings_acceleration" to the fromZigbee section
in "node_modules/zigbee-herdsman-converters/devices.js"



On Tue, 18 Aug 2020 at 02:20, Dejan Zelic <notifications@github.com> wrote:

> @robstonham-personal <https://github.com/robstonham-personal> were you
> able to confirm this as working? I have 2 Samsung Smartthings v4
> multisensors that never toggle the "moving" attribute. The moving attribute
> is recongnised on the devices (and shows in mqtt and home assistant) but
> moving is never triggered to be true (no matter how much I move the
> sensor). I know that these sensors have the accelerometer because they used
> to work with ZHA.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Koenkk/zigbee-herdsman-converters/pull/1415#issuecomment-675195331>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG5TJ3YWE44VUXMUL2YGFFLSBHJPHANCNFSM4PARVEXA>
> .
>

dejanzelic commented:
Thanks @robstonham-personal I created the following PR to fix this: https://github.com/Koenkk/zigbee-herdsman-converters/pull/1498 After making the change you suggested, it started working for me.
